### ISSUE TITLE
Files' size is unknown when read entries in zip 

### ISSUE THREAD

**Issue by Mararsh** [Mararsh/MyBox#478]

**Comment by Mararsh** [comment]
upgrade to Commons Compress 1.19.

**Comment by Mararsh** [comment]
The 1.19 which I download in last version was strange and it missed many files. So I gave up 1.19 and used 1.18.
Today I download 1.19 again. It works well.

**Comment by Mararsh** [comment]
https://commons.apache.org/proper/commons-compress/limitations.html

> ZipArchiveInputStream is limited and may even return false contents in some cases, use ZipFile whenever possible. See the ZIP documentation page for details. This limitation is a result of streaming data vs using random access and not a limitation of Compress' specific implementation.

So ZipFile is the solution

**Comment by Mararsh** [comment]
https://issues.apache.org/jira/browse/COMPRESS-67

``` 
ZIP has a feature that is called the data descriptor.

When your entry uses the DEFLATE method, you don't have to provide the size (and CRC) information inside the local file header because the compression algorithm itself will signal when you've read all data. In that case the size information will be inside the ZIP after the entry's data (and you can check whether the size/CRC matched what you've read).

Unfortunately this means you have to read the full entry in order to provide the size information, which we cannot really do in a stream (without buffering and running the risk of causing OOM exceptions for really big entries). ZipFile can and does provide the correct information.

I don't see any way to fix this, but we certainly have to document it - and list it as another reason prefer ZipFile over ZipArchiveInputStream.

ZipArchiveInputStream will set the correct size and CRC information inside the LFH when it knows it is writing to a file, but it will also use the data descriptor so ZipArchiveInputStream won't trust that information.
```

**Comment by Mararsh** [comment]
Fixed in v5.9
